### PR TITLE
build: Bump minimum required node version to 10.19.0

### PR DESCRIPTION
**Pull request by GChuf**
Critical security vulnerability fixed in node v 10.19.0
Other performance and security updates since node 10.9.0

More info in this article here:
https://snyk.io/blog/node-js-release-fixes-a-critical-http-security-vulnerability/

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   "license": "MIT",
   "//engines-comment": "Keep this in sync with /aio/package.json and /aio/tools/examples/shared/package.json",
   "engines": {
-    "node": ">=10.9.0 <13.0.0",
+    "node": ">=10.20.0 <13.0.0",
```
Comment (gkalpak) [inline package.json:11]: OOC, why not 10.19.0?

**Review by gkalpak — APPROVED** [review]
One question. Otherwise lgtm 🚀
Thx, @GChuf :+1:

File: package.json (hunk 1 above)
Comment (GChuf) [reply package.json:11]: 10.20. has some other notable changes, like updating to ssl 1.1.1e, and is built on a newer macOS version to support package notarization. I'd much rather bump the version to the newest one (10.24), but I think that release might be "too recent" ... 10.20. was published in april last year.

Comment (gkalpak) [reply package.json:11]: I just don't see a the reason to specifically force >=10.20.0. For >=10.19.0 it makes sense due to the critical security vulnerability that was fixed. Do I miss something?

Comment (GChuf) [reply package.json]: "reverted" to 10.19.0.

**Review by gkalpak — APPROVED** [review]

**Comment by clydin** [comment]
Node.js v10 reaches its end of life at the end of this month (April 30, 2021).  If the version is going to be increased it should probably be to a v12 version.

**Comment by alan-agius4** [comment]
We should also update https://github.com/angular/angular/blob/4ce743dfb8b89b54093c4c9300ca24c5d1f50544/packages/compiler-cli/package.json#L35

**Comment by gkalpak** [comment]
Good points, @clydin and @alan-agius4 :+1: Thx for bringing these up.
This PR is fine as it is imo. We should merge this and then discuss what versions/ranges we want to update to and when.

**Pull request closed without merging**